code/script.js

```diff
@@ -220,106 +224,187 @@ const generateBoard = () => {
       </div>
     `
   })
+  console.log(generateBoard)
 }
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
 
-// This function to start (and restart) the game
-const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
 
-// setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
 
+// Setting the currentQuestion object when you select something in the dropdown
+const selectQuestion = () => {
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
-  currentQuestion = {
-    category: category,
-    // value: value
-  }
+  const category = questions.options[questions.selectedIndex].parentNode.label 
+  // This variable stores the actual value of the question we've selected.
+  const value = questions.value
+  
+  if (category === 'hair') {
+    currentQuestion = {
+      category: 'hair',
+      value: value
+    }
+  } else if (category === 'eyes') {
+    currentQuestion = {
+      category: 'eyes',
+      value: value
+    }
+  } else if (category === 'accessories') {
+    currentQuestion = {
+      category: 'accessories',
+      value: value
+    }
+  } else {
+    currentQuestion = {
+      category: 'other',
+      value: value
+    }
+  }  
+  // This is an object that stores the key: and its value
+  //currentQuestion = {
+  // category: category, // hair, eyes or accessories
+  // value: value     // blue or brown etc
+  // }
 }
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
+  const { category, value } = currentQuestion; // This is to destruct the currentQuestion object to be able to use these variables more easily
+  if (category === 'eyes') {
```

Comment:
To clean up and shorten the code a bit, the eyes and hair could be written the same way as you've done for accessories and other